src/SW_Flow.c

```diff
@@ -224,6 +224,11 @@ void SW_Water_Flow(SW_ALL* sw, LOG_INFO* LogInfo) {
 		);
 	}
 
+	ForEachSoilLayer(i, n_layers) {

```

Comment:
It seems that setting the array `sw->SoilWat.avgLyrTemp[]` to 0 on each day of the simulation is unnecessary for most cases and expensive. We only need to `sw->SoilWat.avgLyrTemp[]` initialize once during a simulation run and only if the soil temperature module is turned off.

Here is my suggestion to solve this issue: https://github.com/DrylandEcology/SOILWAT2/pull/354